rcl/src/rcl/wait.c

```diff
@@ -324,13 +325,14 @@ rcl_wait_set_get_allocator(const rcl_wait_set_t * wait_set, rcl_allocator_t * al
   /* Also resize the rmw storage. */ \
   wait_set->impl->RMWCount = 0; \
   wait_set->impl->RMWStorage = (void **)allocator.reallocate( \
-    wait_set->impl->RMWStorage, sizeof(rcl_ ## Type ## _t *) * size, allocator.state); \
+    wait_set->impl->RMWStorage, sizeof(rmw_ ## Type ## _t *) * size, allocator.state); \
```

Comment:
Not sure this is the right type still:
`wait_set->impl->RMWStorage` is something like `wait_set->impl->rmw_subscriptions.subscribers` which [has type `void **`](https://github.com/ros2/rmw/blob/ac8ccc7b81f718863417783fa372818e682bd707/rmw/include/rmw/types.h#L97); we don't seem to store the whole `rmw_subscription_t` struct in `wait_set->impl->RMWStorage`, but [only the `data` field](https://github.com/ros2/rcl/blob/7d0045adb847ba55419198160eb74d381ac8e9c4/rcl/src/rcl/wait.c#L266), which has type `void *`.

Reply:
Thanks for double checking :+1: , I had a doubt when reviewing the original change. `void *` seems to be the right type indeed